### PR TITLE
Added small implementation of DavFolder

### PR DESCRIPTION
**Pull request by deinok**

### CODE REVIEW

File: src/main/java/com/github/sardine/impl/SardineImpl.java [hunk 1]
```diff
@@ -107,11 +57,7 @@
 import org.apache.http.entity.InputStreamEntity;
 import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.auth.BasicScheme;
-import org.apache.http.impl.client.BasicAuthCache;
-import org.apache.http.impl.client.BasicCredentialsProvider;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.client.*;
```
Comment (lookfirst) [inline src/main/java/com/github/sardine/impl/SardineImpl.java]: No star imports please. =)

File: src/test/java/com/github/sardine/DavFolderTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Before
+    public void before() throws IOException {
+        this.properties = new Properties();
+        this.sardine = SardineFactory.begin("jgg15", "warhammer2611");
+    }
+
+    @Test
+    public void redirectTest1() throws IOException, URISyntaxException {
+        DavFolder davFolder = this.sardine.getRoot("https://cv.udl.cat/dav/102006-1516/");
```
Comment (lookfirst) [inline src/test/java/com/github/sardine/DavFolderTest.java]: I'd rather not have tests hitting a remote server. We are moving away from this.

**Comment by deinok** [comment]
Okey, sorry, IntelliJ do have done it.
Next day i will fix it ;)

**Comment by dkocher** [comment]
I don't see why this (recursing?) operation should be implemented in the library.

**Comment by lookfirst** [comment]
@dkocher I agree, however if he wants to show us his source code, that is fine.

**Comment by deinok** [comment]
@dkocher @lookfirst What if you want to have a map of the WebDav Server, and not a list.

I'm developing an app that should show to the users the file system of the server. I think its a good idea have the ability to access it like this.

If is there any other methodology to get the same, please contact me and i will close the PR;)

**Comment by lookfirst** [comment]
@deinok Sardine is used as the basis for the amazing app CyberDuck. Somehow @dkocher was able to make it work just fine. ;-)

**Comment by lookfirst** [comment]
More code in Sardine is more complexity, more complexity is bad. =) Sardine is meant to be a library that makes it easy to build features like what you've built, but that doesn't mean that it should go into Sardine directly. I appreciate your contribution and desire, but I agree with @dkocher that we shouldn't include this.

**Pull request closed without merging**